app/gui/src/project-view/components/GraphEditor/GraphNode.vue

```diff
@@ -257,6 +263,16 @@ watch(isVisualizationPreviewed, (newVal, oldVal) => {
   }
 })
 
+const scale = computed(() => navigator?.scale ?? 1)
+provideResizableWidgetRegistry(
+  computed({
+    get: () => visualizationWidth.value && visualizationWidth.value * scale.value,
+    set: (width) => (visualizationWidth.value = width && width / scale.value),
+  }),
+  () => CONTENT_PADDING * scale.value,
+  () => widgetTreeSize.value.x,
+)
+
```

Comment:
I think this may cause a performance problem in large graphs--adding a reactive dependency on `navigator.scale` will require Vue to update every `GraphNode` every frame while zooming the graph. Wouldn't it be more efficient for the registry to use scene coordinates?

Reply:
GraphNode components should not be updated; `nodeWidth` change in resizableWidgetRegistry will trigger a watch calling `adjustToNodeWidth`, but here we would just compute exactly the same widget size, which is discarded before writting to metadata.

And because I rely on `useResizeObserver`, the code would be launched anyway on zoom in/out. The optimized solution would require a completely different solution. I assume what is so far is enough - but I test it with 100 nodes yet.

Reply:
I tested it on 100 Table.input nodes. It's laggy, but same way as on develop. Interestingly, AgGrid is refreshed there when zooming - probably worth checking at some point.